compose/service.py

```diff
@@ -241,29 +245,15 @@ def recreate_container(self, container, **override_options):
             else:
                 raise
 
-        intermediate_container = Container.create(
-            self.client,
-            image=container.image,
-            entrypoint=['/bin/echo'],
-            command=[],
-            detach=True,
-            host_config=create_host_config(volumes_from=[container.id]),
-        )
-        intermediate_container.start()
-        intermediate_container.wait()
-        container.remove()
-
-        options = dict(override_options)
+        # Use a hopefully unique container name by prepending the short id
+        self.client.rename(container.id, container.short_id + container.name)
```

Comment:
Tiny thing, but right now this generates names like `c8a6039dcecounter_web_1`. Could we stick an underscore between the short ID and the name?